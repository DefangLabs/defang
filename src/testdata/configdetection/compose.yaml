services:
  configdetection: 
    image: alpine
    environment:
      - API_KEY=50m34p1k3y # keyword detector
      - AWS_CLIENT_ID=AROA1234567890ABCDEF # aws_client_id detector
      - GH_PAT=ghp_aBcDeFgHiJkLmNoPqRsTuVwXyZ1234567890 # github detector
      - HIGH_ENTROPY_STRING=1234567890abcdef # high_entropy_string detector
      - MY_URL=https://user:p455w0rd@example.com # url_password detector
      - NOT_A_CONFIG=notaconfigvalue