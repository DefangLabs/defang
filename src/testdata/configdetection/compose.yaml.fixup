{
  "configdetection": {
    "command": null,
    "entrypoint": null,
    "environment": {
      "API_KEY": "50m34p1k3y",
      "HIGH_ENTROPY_STRING": "1234567890abcdef",
      "MY_URL": "https://user:p455w0rd@example.com"
    },
    "image": "alpine",
    "networks": {
      "default": null
    }
  }
}