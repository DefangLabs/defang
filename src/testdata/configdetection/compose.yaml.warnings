 ! service "configdetection": environment variable "API_KEY" may contain sensitive values. Consider using 'defang config set API_KEY'
 ! service "configdetection": environment variable "AWS_CLIENT_ID" may contain sensitive values. Consider using 'defang config set AWS_CLIENT_ID'
 ! service "configdetection": environment variable "GH_PAT" may contain sensitive values. Consider using 'defang config set GH_PAT'
 ! service "configdetection": environment variable "HIGH_ENTROPY_STRING" may contain sensitive values. Consider using 'defang config set HIGH_ENTROPY_STRING'
 ! service "configdetection": environment variable "MY_URL" may contain sensitive values. Consider using 'defang config set MY_URL'
 ! service "configdetection": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors