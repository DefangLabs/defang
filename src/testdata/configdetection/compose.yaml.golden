name: configdetection
services:
  configdetection:
    environment:
      API_KEY: 50m34p1k3y
      AWS_CLIENT_ID: AROA1234567890ABCDEF
      GH_PAT: ghp_aBcDeFgHiJkLmNoPqRsTuVwXyZ1234567890
      HIGH_ENTROPY_STRING: 1234567890abcdef
      MY_URL: https://user:p455w0rd@example.com
      NOT_A_CONFIG: notaconfigvalue
    image: alpine
    networks:
      default: null
networks:
  default:
    name: configdetection_default
